### ISSUE TITLE
False negative for the DL4006 rule

### ISSUE THREAD

**Issue by tymonx** [zabio3/godolint#16]
Some CLI tools use pipe `|` in their command line arguments. For example the [yq](https://github.com/mikefarah/yq) tool for parsing and handling YAML files.

This simple example fails even if the shell pipe `|` is nowhere used:

```Dockerfile
RUN yq eval '.".docker".script[1] | explode(.)' /base.yml > /usr/local/bin/entrypoint
```

Error:

```plaintext
DL4006 Set the `SHELL` option -o pipefail before `RUN` with a pipe in it.
```
